runtime/autoload/provider/pythonx.vim

```diff
@@ -13,25 +13,36 @@ function! provider#pythonx#Detect(ver) abort
   let skip = exists(skip_var) ? {skip_var} : 0
   if exists(host_var)
     " Disable auto detection
-    let [check, err] = s:check_interpreter({host_var}, a:ver, skip)
-    return check ? [{host_var}, err] : ['', err]
+    let [chk, err] = s:check_interpreter({host_var}, a:ver, skip)
+    return chk ? [{host_var}, err] : ['', err]
   endif
 
   let detect_versions = (a:ver == 2) ?
-        \   ['2.7', '2.6', '2', '']
-        \ : ['3.5', '3.4', '3.3', '3.2', '3', '']
+        \   ['2', '']
+        \ : ['3', '']
```

Comment:
I think your fix should be merged in https://github.com/neovim/neovim/pull/2549.
And please consider about https://github.com/neovim/neovim/pull/1334#issuecomment-94178512 problem.